service_configuration_lib/spark_config.py

```diff
@@ -929,31 +911,44 @@ def _append_event_log_conf(
 
         if len(self.spark_srv_conf.items()) == 0:
             log.warning('spark_srv_conf is empty, disable event log')
+            spark_opts.update({"spark.eventLog.enabled": "false"})
             return spark_opts
 
-        try:
-            account_id = (
-                boto3.client(
-                    'sts',
-                    aws_access_key_id=access_key,
-                    aws_secret_access_key=secret_key,
-                    aws_session_token=session_token,
+        if access_key:
+            try:
+                account_id = (
+                    boto3.client(
+                        'sts',
+                        aws_access_key_id=access_key,
+                        aws_secret_access_key=secret_key,
+                        aws_session_token=session_token,
+                    )
+                    .get_caller_identity()
+                    .get('Account')
                 )
-                .get_caller_identity()
-                .get('Account')
-            )
-        except Exception as e:
-            log.warning('Failed to identify account ID, error: {}'.format(str(e)))
-            spark_opts['spark.eventLog.enabled'] = 'false'
-            return spark_opts
+            except Exception as e:
+                log.warning('Failed to identify account ID, error: {}'.format(str(e)))
+                spark_opts['spark.eventLog.enabled'] = 'false'
+                return spark_opts
+
+            for conf in self.spark_srv_conf.get('environments', {}).values():
+                if account_id == conf['account_id']:
+                    spark_opts['spark.eventLog.enabled'] = 'true'
+                    spark_opts['spark.eventLog.dir'] = conf['default_event_log_dir']
```

Comment:
Currently we are reading `default_event_log_dir` from srv_configs and set `spark.eventLog.enabled`, `spark.eventLog.dir` programmatically. Since we don't support custom eventLog dir, can we move these two Spark configs to `mandatory_defaults` in srv_configs to be loaded together with other static Spark configs?

Reply:
But this config is not static because it needs to identify `account_id` or `environment` to choose the correct one dynamically. We have only put static ones in `mandatory_defaults`